src/main/java/Duke.java

```diff
@@ -1,10 +1,141 @@
+import java.util.Scanner;
+import java.util.Vector;
+
 public class Duke {
+    public class Task {
+        protected String description;
+        protected boolean isDone;
+        protected String taskType;
+        protected String taskDate;
+    
+        public Task(String description,boolean isDone, String taskType, String taskDate) { // initiating an instance
+            this.description = description;
+            this.isDone = isDone;
+            this.taskType = taskType; // todo,ddl,event
+            this.taskDate = taskDate; // 
+
+        }
+
+        public String getTaskType(){
+            return (this.taskType);
+        }
+    
+        public String getStatusIcon() {
+            return (isDone ? "X" : " "); // mark done task with X
+        }
+    
+        public void markAsDone(){
+            this.isDone = true;
+        }
+
+        public void markAsNotDone(){
+            this.isDone = false;
+        }
+
+        public String getDescription(){
+            return this.description;
+
+        }
+
+        public String getDate(){
+            return this.taskDate;
+    }
+}
+
+
+    // Driver code
     public static void main(String[] args) {
+        // Greeting message
         String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
+                    + "|  _ \\ _   _| | _____ \n"
+                    + "| | | | | | | |/ / _ \\\n"
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println("Hello! I'm Duke_HTT. \n What can I do for you?");
```

Comment:
Might want to consider separating this out as a method to make the code cleaner and more modular, especially since a substantially large graphic is being printed.